File: api-core/src/main/java/io/nerv/core/bizlog/supporter/jdbc/JDBCSupporter.java
package io.nerv.core.bizlog.supporter.jdbc;

import io.nerv.core.bizlog.base.BizLogEntity;
import io.nerv.core.bizlog.base.BizLogSupporter;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.jdbc.core.JdbcTemplate;

import java.util.Date;
import java.util.List;

/**
 * 基于数据库的日志持久话类
 * @author: S.PKAQ
 * @Datetime: 2018/9/26 22:09
 */
public class JDBCSupporter implements BizLogSupporter {
    private BizLogEntity bizLogEntity;

    public JDBCSupporter(BizLogEntity bizLogEntity) {
        this.bizLogEntity = bizLogEntity;
    }

    @Autowired
    private JdbcTemplate jdbcTemplate;

    @Override
    public void save(BizLogEntity entity) {

    }

    @Override
    public List<BizLogEntity> getLog() {
        return null;
    }

    @Override
    public List<BizLogEntity> getLogByType(String type) {
        return null;
    }

    @Override
    public List<BizLogEntity> getLogAfter(Date dateTime) {
        return null;
    }

    @Override
    public List<BizLogEntity> getLogBetween(Date begin, Date end) {
        return null;
    }

    @Override
    public void cleanAll() {

    }

    @Override
    public void cleanBefore(Date dateTime) {

    }

    @Override
    public void cleanBetween(Date begin, Date end) {

    }

    @Override
    public void print() {

    }
}

File: eva-core/src/main/kotlin/io/nerv/core/bizlog/supporter/jdbc/JDBCSupporter.kt
package io.nerv.core.bizlog.supporter.jdbc

import io.nerv.core.bizlog.base.BizLogEntity
import io.nerv.core.bizlog.base.BizLogSupporter
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.jdbc.core.JdbcTemplate
import java.util.*

/**
 * 基于数据库的日志持久话类
 * @author: S.PKAQ
 * @Datetime: 2018/9/26 22:09
 */
class JDBCSupporter(private val bizLogEntity: BizLogEntity) : BizLogSupporter {

    @Autowired
    private val jdbcTemplate: JdbcTemplate? = null
    override fun save(entity: BizLogEntity?) {}
    override val log: List<BizLogEntity?>?
        get() = null

    override fun getLogByType(type: String?): List<BizLogEntity?>? {
        return null
    }

    override fun getLogAfter(dateTime: Date?): List<BizLogEntity?>? {
        return null
    }

    override fun getLogBetween(begin: Date?, end: Date?): List<BizLogEntity?>? {
        return null
    }

    override fun cleanAll() {}
    override fun cleanBefore(dateTime: Date?) {}
    override fun cleanBetween(begin: Date?, end: Date?) {}
    override fun print() {}

}